### PR TITLE
feat(argo-workflows): Allow NodePort to be specified for server service.

### PR DESCRIPTION
**Pull request by peteroruba**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by peteroruba** [comment]
@mkilchhofer how do I run the fifth DCO action (build is green)? Didn't find any hint in the documentation. Thanks

File: charts/argo-workflows/templates/server/server-service.yaml [hunk 1]
```diff
@@ -17,6 +17,9 @@ spec:
     name: {{ . }}
     {{- end }}
     targetPort: 2746
+    {{- if eq .Values.server.serviceType "NodePort" }}
```
Comment (mkilchhofer) [inline charts/argo-workflows/templates/server/server-service.yaml:20]: Let's use the same conditional as we use for LoadBalancer some lines below:
```suggestion
    {{- if and (eq .Values.server.serviceType "NodePort") .Values.server.serviceNodePort }}
```

File: charts/argo-workflows/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: argo-workflows
 description: A Helm chart for Argo Workflows
 type: application
-version: 0.6.0
+version: 0.6.1
```
Comment (mkilchhofer) [inline charts/argo-workflows/Chart.yaml:5]: As this adds a feature, please bump the minor version.
```suggestion
version: 0.7.0
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-workflows/templates/server/server-service.yaml (hunk 1 above)
Comment (peteroruba) [reply charts/argo-workflows/templates/server/server-service.yaml]: Resolved, as per PR update

File: charts/argo-workflows/Chart.yaml (hunk 2 above)
Comment (peteroruba) [reply charts/argo-workflows/Chart.yaml]: Resolved, as per PR update

**Review by mkilchhofer — APPROVED** [review]
Lgtm

**Comment by peteroruba** [comment]
Thank you for reviewing my PR @mkilchhofer

**Pull request merged**